In [ ]:
from pathlib import Path
import pandas as pd

LOCAL = Path("final_project_model_results_summary.csv")
REPO  = Path("results/final_project_model_results_summary.csv")

path = LOCAL if LOCAL.exists() else REPO
print(f"Reading from: {path}")
df = pd.read_csv(path)
df.head()

Читаю з: final_project_model_results_summary.csv


Model     F1  Accuracy  Precision  Recall  \
0  TF_IDF + Logistic Regression  0.986     0.986      0.986   0.986   
1     BOW + Logistic Regression  0.996     0.996      0.996   0.996   
2        TF-IDF + SVD + XGBoost  0.983     0.983      0.983   0.983   
3                          Bert  0.999     0.999      0.999   0.999   

                                     Notebook  
0  Fake_news_preprocessing_baseline_XGB.ipynb  
1  Fake_news_preprocessing_baseline_XGB.ipynb  
2  Fake_news_preprocessing_baseline_XGB.ipynb  
3                        Fake_news_BERT.ipynb

## Final Conclusion on Models

**Ranking by F1:**
1) **BERT** — F1 **0.999**, Acc 0.999 → best quality.  
2) **BoW + Logistic Regression** — F1 **0.996**, Acc 0.996 (–0.003 vs BERT) → almost as good as BERT, but much lighter for inference.  
3) **TF-IDF + Logistic Regression** — F1 **0.986**, Acc 0.986 (–0.013) → reliable, interpretable baseline.  
4) **TF-IDF + SVD + XGBoost** — F1 **0.983**, Acc 0.983 (–0.016) → in this configuration performs worse than linear models.  

**Notes on reliability:**
- Duplicates across train/val/test were removed, `date` column dropped → no leakage, metrics are valid.  
- Near-perfect scores (~1.0) should be double-checked — run cross-validation or test on a separate (new) dataset.  

**Next steps:**
- Expand **error analysis** (short headlines, sarcasm, domain shifts).  
- TF-IDF + SVD + XGBoost: try tuning SVD size (e.g., 300/500) and basic XGBoost parameters (`max_depth`, `n_estimators`, `learning_rate`) with fixed seed.  
- Stability check: run stratified cross-validation (e.g., 5-fold) or test on a new dataset.  
